org-node/src/webserver.rs

```diff
@@ -0,0 +1,170 @@
+// |   Process      |     |                                     |     |
+// |   (Spawned)    |     |  Sender<WsEvent>                    |     |
+// |                |     |  Sender<EstablishWebSocketSender>   |     |
+// +----------------+     |                                     |     |
+//                        |  +----------------------------------v--+  |
+//                        |  | Connected Clients (Spawned)         |  |
+//                        |  |                                     |  |
+//                        |  | HashMap<SocketAddr,Sender<WsEvent>> |  |
+//                        |  | Receiver<WsEvent>                   |  |
+//                        |  | Receiver<EstablishWebSocketSender>  |  |
+//                        |  +-----^-+------------+-------------+--+  |
+//                        |        | |            |             |     |
+//                        +---+----+-+------+-----+---------+---+-----+
+//                            |      |      |     |         |   |
+//                        +---+------v+  +--+-----v---+  +--+---v-----+
+//                        |           |  |            |  |            |
+//                        | WS Client |  | WS Client  |  | WS Client  |
+//                        +-----------+  +------------+  +------------+
+
+/// Websocket server endpoint, e.g. ws://0.0.0.0:8336/subscribe
+pub const WEBSOCKET_PATH: &str = "subscribe";
+
+/// WebSocket event enum type to broadcast to connected websocket peers.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum WsEvent {
+    UpdatedRef {
+        urn: Urn,
+        // Rev is Rev::Git(git2::Oid)
+        // Using string to allow for serialization
+        rev: String,
+        peer: PeerId,
+    },
+}
+
+/// Establishes a websocket connection and sends new connected client event to message bus
+/// to update connected websocket client mapping to broadcast events to connected clients.
+async fn establish_connection(
+    websocket: WebSocket,
+    remote: Option<SocketAddr>,
+    conn_tx: mpsc::UnboundedSender<EstablishWebSocketSender>,
+) {
+    // Map the websocket stream to the channel
+    tracing::debug!(
+        target: "org-node",
+        "Received connection request from peer"
+    );
+
+    let (mut peer_ws_tx, mut peer_ws_rx) = websocket.split();
+
+    // Use an unbounded channel to handle buffering and flushing of messages.
+    let (peer_unbounded_tx, peer_unbounded_rx) = mpsc::unbounded_channel::<WsEvent>();
+    let mut peer_unbounded_rx = UnboundedReceiverStream::new(peer_unbounded_rx);
+
+    // Listen for internal events and send to connected client.
+    spawn(async move {
+        while let Some(message) = peer_unbounded_rx.next().await {
+            if let Ok(msg) = serde_json::to_string(&message) {
+                peer_ws_tx
+                    .send(Message::text(msg))
+                    .unwrap_or_else(|e| {
+                        eprintln!("websocket send error: {}", e);
```

Comment:
`tracing::error`

Reply:
Not sure what happened there... 😵 

commit 0452b3c fixes.